Add Omnibus definition for aerospike

ofek opened the pull request:
### Motivation

The binary wheels on PyPI are not yet compatible with OpenSSL 1.1.0+

ofek committed:
Add Omnibus definition for aerospike

KSerrania commented on omnibus/config/software/datadog-agent-integrations-py2.rb:
```diff
@@ -28,6 +28,10 @@
   # need kerberos for hdfs
   dependency 'libkrb5'
+
+  unless suse || arm?
```
```suggestion
  unless suse? || arm?
```

KSerrania commented on omnibus/config/software/datadog-agent-integrations-py3.rb:
```diff
@@ -28,6 +28,10 @@
   # need kerberos for hdfs
   dependency 'libkrb5'
+
+  unless suse || arm?
```
```suggestion
  unless suse? || arm?
```

ofek committed:
Apply suggestions from code review

Co-Authored-By: Kylian Serrania <kylian.serrania@datadoghq.com>

ofek committed:
fix

ofek committed:
lua too

ofek committed:
try

ofek committed:
split def

ofek committed:
fix

ofek committed:
final

ofek committed:
refactor

KSerrania committed:
[omnibus] Fix aerospike_client software definition and cleanup aerospike build (#4955)

Uses EXT_CFLAGS instead of CFLAGS during aerospike C client build.
Builds the C client outside of the install_dir to avoid shipping it (it's not needed at runtime).
Uses omnibus' git fetcher feature.

KSerrania committed:
Merge branch 'master' into ofek/a

albertvaka reviewed (approved):
Thanks for working on this guys :)

Are we okay with not having aerospike on suse, though?

KSerrania commented:
@albertvaka it was already like this before, the definition was temporarily changed in #4954 to blacklist it on all platforms instead of SUSE and ARM only.
